src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,446 @@
+                            {
+                                var localResult = analysisResult.TryGetInterproceduralResultByDefinition(containingSymbol);
+                                if (localResult != null)
+                                {
+                                    var localValue = localResult[platformSpecificOperation.Key.Kind, platformSpecificOperation.Key.Syntax];
+                                    if (localValue.Kind == GlobalFlowStateAnalysisValueSetKind.Known)
+                                    {
+                                        if (IsKnownValueGuarded(attribute, localValue))
+                                        {
+                                            continue;
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                        else if (value.Kind == GlobalFlowStateAnalysisValueSetKind.Known)
+                        {
+                            if (IsKnownValueGuarded(attribute, value))
+                            {
+                                continue;
+                            }
+                        }
+
+                        ReportDiagnostics(platformSpecificOperation.Key, attribute, context);
+                    }
+                }
+                finally
+                {
+                    platformSpecificOperations.Free();
+                }
+
+                return;
+
+                OperationVisitor CreateOperationVisitor(GlobalFlowStateAnalysisContext context) => new OperationVisitor(guardMethods, osPlatformType, context);
+            });
+        }
+
+        private static bool IsKnownValueGuarded(PlatformAttributeInfo attribute, GlobalFlowStateAnalysisValueSet value)
+        {
+            foreach (var analysisValue in value.AnalysisValues)
+            {
+                if (analysisValue is RuntimeMethodValue info)
+                {
+                    if (!info.Negated)
+                    {
+                        if (IsOSPlatformsEqual(attribute.PlatformName, info.PlatformPropertyName))
+                        {
+                            if (info.InvokedPlatformCheckMethodName == s_platformCheckMethodNames[0])
+                            {
+                                if (attribute.AttributeType == PlatformAttributeType.MinimumOSPlatformAttribute &&
+                                    AttributeVersionsMatch(attribute.AttributeType, attribute.Version, info.Version))
+                                {
+                                    return true;
+                                }
+                            }
+                            else
+                            {
+                                if ((attribute.AttributeType == PlatformAttributeType.ObsoletedInOSPlatformAttribute ||
```

Comment:
Do we have different functionality anywhere in the analyzer between `ObsoletedInOSPlatformAttribute` and `RemovedInOSPlatformAttribute`? Otherwise, can consider just removing one of the redundant `PlatformAttributeType` fields for them.

Reply:
With the current requirement, they are same, good point